### ISSUE TITLE
Github Copilot tool window is not read by NVDA screen reader

### ISSUE THREAD

**Issue by heiko-folkerts-msg-david** [microsoft/copilot-intellij-feedback#256]
The tool window "TGithub Copilot" cannot be used with screen reader NVDA. The focus lands somewhere in the window hierarchy. If a sighted assistend clicks into the window it seemed to be usable.
This bug only applies to the window "github Copilot" but not to "github copilot chast". The chat window is very accessible!

- Versions (plugin, IDE, OS):

- Debug-level Log Files:

Steps to Reproduce:

1.
2.

Screenshots/Recording:

**Comment by Eskibear** [comment]
The "GitHub Copilot" panel was designed for completion when there was no inline completion APIs. Today as JetBrains already offers that API and the Copilot plugin is implemented based on that. 
Before we fix the accessibility of the "GitHub Copilot" panel, are you able to use inline completion (ghost text when you type code in the editor)?

**Comment by heiko-folkerts-msg-david** [comment]
Hi,
unfortunately „getting inline text“ for NVDA users means that a addon for NVDA reads out the text proposed. It seems that this addon is no longer compatible with the current version of IntelliJ.
So I cannot confirm that.
But if the panel is not really needed, then this is an important information for me.


Mit freundlichen Grüßen
Dipl.-Inf. (FH) Heiko Folkerts
DevOps Engineer
msg for automotive gmbh

Mittelweg 7
38106 Braunschweig

Tel.: +49 89 96 101 15 10

Mobil: +49 152 22641316
E-Mail: ***@***.***
https://www.msg.group<https://www.msg.group/de/branchen/automotive>
Sitz der Gesellschaft: 85737 Ismaning
Geschäftsführer: Martin Ober, Dr. Andreas Scholz
Handelsregister: Amtsgericht München, HRB 295920

Bitte beachten Sie, dass der Inhalt dieser E-Mail vertraulich zu behandeln ist. Sofern Sie nicht der beabsichtigte Adressat sind, dürfen wir Sie bitten, uns umgehend zu benachrichtigen und den Inhalt dieser E-Mail zu löschen.
[msg – creating value for automotive  We lead the way in digital transformation.]

Von: Yan Zhang ***@***.***>
Gesendet: Mittwoch, 28. Mai 2025 09:12
An: microsoft/copilot-intellij-feedback ***@***.***>
Cc: Heiko Folkerts ***@***.***>; Author ***@***.***>
Betreff: Re: [microsoft/copilot-intellij-feedback] Github Copilot tool window is not read by NVDA screen reader (Issue #256)


Caution: This email originated from outside of the organization. Despite an upstream security check of attachments and links by Microsoft Defender for Office, a residual risk always remains. Only open attachments and links from known and trusted senders.
[Das Bild wurde vom Absender entfernt.]Eskibear left a comment (microsoft/copilot-intellij-feedback#256)<https://github.com/microsoft/copilot-intellij-feedback/issues/256#issuecomment-2915224389>

The "GitHub Copilot" panel was designed for completion when there was no inline completion APIs. Today as JetBrains already offers that API and the Copilot plugin is implemented based on that.
Before we fix the accessibility of the "GitHub Copilot" panel, are you able to use inline completion (ghost text when you type code in the editor)?

**Comment by Eskibear** [comment]
OK, so if inline completion is not working for NVDA users, I think we need to fix the accessibility issue of the completion panel. 

> The focus lands somewhere in the window hierarchy. 

Can you share more details about that? Are you able to focus on the panel? Or it just reads unexpected text?

**Comment by heiko-folkerts-msg-david** [comment]
Hello Yan Zhang,
When activating the github copilot tool window the focus lands on a sibling of the log window. Unfortunately NVDA shows no trackable information for that window. Only window handle and other dynamic things but e.g. no control id. Pressing tab has no effect and pressing Shift+tab brings me back into my editor.
Something else I could try?

Mit freundlichen Grüßen
Dipl.-Inf. (FH) Heiko Folkerts
DevOps Engineer
msg for automotive gmbh

Mittelweg 7
38106 Braunschweig

Tel.: +49 89 96 101 15 10

Mobil: +49 152 22641316
E-Mail: ***@***.***
https://www.msg.group<https://www.msg.group/de/branchen/automotive>
Sitz der Gesellschaft: 85737 Ismaning
Geschäftsführer: Martin Ober, Dr. Andreas Scholz
Handelsregister: Amtsgericht München, HRB 295920

Bitte beachten Sie, dass der Inhalt dieser E-Mail vertraulich zu behandeln ist. Sofern Sie nicht der beabsichtigte Adressat sind, dürfen wir Sie bitten, uns umgehend zu benachrichtigen und den Inhalt dieser E-Mail zu löschen.
[msg – creating value for automotive  We lead the way in digital transformation.]

Von: Yan Zhang ***@***.***>
Gesendet: Donnerstag, 29. Mai 2025 19:59
An: microsoft/copilot-intellij-feedback ***@***.***>
Cc: Heiko Folkerts ***@***.***>; Author ***@***.***>
Betreff: Re: [microsoft/copilot-intellij-feedback] Github Copilot tool window is not read by NVDA screen reader (Issue #256)


Caution: This email originated from outside of the organization. Despite an upstream security check of attachments and links by Microsoft Defender for Office, a residual risk always remains. Only open attachments and links from known and trusted senders.
[Das Bild wurde vom Absender entfernt.]Eskibear left a comment (microsoft/copilot-intellij-feedback#256)<https://github.com/microsoft/copilot-intellij-feedback/issues/256#issuecomment-2920158502>

OK, so if inline completion is not working for NVDA users, I think we need to fix the accessibility issue of the completion panel.

The focus lands somewhere in the window hierarchy.

Can you share more details about that? Are you able to focus on the panel? Or it just reads unexpected text?
